### PR TITLE
Fix a undefined behavior in `ExtractContracts.cpp`

### LINKED ISSUE #724: Undefined behavior in `ExtractContracts.cpp`
See:

https://github.com/smackers/smack/blob/465280b2433d3b61d1ebfc86f4ffb39ad0c047f3/lib/smack/ExtractContracts.cpp#L95

`LI.getLoopFor(B)` could be a nullptr if `B` is not in `LI`. We haven't hit a crash probably because it's a method call that doesn't access `this`.

### PR DESCRIPTION
**Pull request by shaobo-he**
Fixes #724

### CODE REVIEW

File: lib/smack/ExtractContracts.cpp [hunk 1]
```diff
@@ -92,10 +92,12 @@ std::tuple<BlockList, LoopMap> splitContractBlocks(Function &F, LoopInfo &LI) {
                           << "\n");
             BasicBlock *B = CI->getParent();
             auto NewBB = B->splitBasicBlock(++CI->getIterator());
-            LI.getLoopFor(B)->addBasicBlockToLoop(NewBB, LI);
+            auto L = LI.getLoopFor(B);
+            if (L)
```
Comment (zvonimir) [inline lib/smack/ExtractContracts.cpp:96]: Should this be an assertion or an if condition? Is the code going to work properly if L is null?

**Review by zvonimir — CHANGES_REQUESTED** [review]

File: lib/smack/ExtractContracts.cpp (hunk 1 above)
Comment (shaobo-he) [reply lib/smack/ExtractContracts.cpp:96]: It should be an if condition because `B` may not be in any loop.

@michael-emmi could you confirm this?

Comment (zvonimir) [reply lib/smack/ExtractContracts.cpp:96]: I double-checked this. @shaobo-he is right and I think this is the correct fix.

**Review by zvonimir — APPROVED** [review]
Looks good to me.

**Pull request merged**